internal/testutils/broker.go

```diff
@@ -0,0 +1,307 @@
+	}
+
+	sessionID, encryptionKey, err := b.broker.NewSession(context.Background(), username, lang)
+	if err != nil {
+		return "", "", dbus.MakeFailedError(err)
+	}
+	return sessionID, encryptionKey, nil
+}
+
+// GetAuthenticationModes calls the broker mock corresponding method and returns the result or an error if requested.
+func (b *BrokerBusMock) GetAuthenticationModes(sessionID string, supportedUILayouts []map[string]string) (authenticationModes []map[string]string, dbusErr *dbus.Error) {
+	if sessionID == "GAM_bus_error" {
+		return nil, dbus.MakeFailedError(errors.New("erroring out as requested"))
+	}
+
+	authenticationModes, err := b.broker.GetAuthenticationModes(context.Background(), sessionID, supportedUILayouts)
+	if err != nil {
+		return nil, dbus.MakeFailedError(err)
+	}
+	return authenticationModes, nil
+}
+
+// SelectAuthenticationMode calls the broker mock corresponding method and returns the result or an error if requested.
+func (b *BrokerBusMock) SelectAuthenticationMode(sessionID, authenticationModeName string) (uiLayoutInfo map[string]string, dbusErr *dbus.Error) {
+	if sessionID == "SAM_bus_error" {
+		return nil, dbus.MakeFailedError(errors.New("erroring out as requested"))
+	}
+
+	uiLayoutInfo, err := b.broker.SelectAuthenticationMode(context.Background(), sessionID, authenticationModeName)
+	if err != nil {
+		return nil, dbus.MakeFailedError(err)
+	}
+	return uiLayoutInfo, nil
+}
+
+// IsAuthorized calls the broker mock corresponding method and returns the result or an error if requested.
+func (b *BrokerBusMock) IsAuthorized(sessionID, authenticationData string) (access, infoUser string, dbusErr *dbus.Error) {
+	if sessionID == "IA_bus_error" {
+		return "", "", dbus.MakeFailedError(errors.New("erroring out as requested"))
+	}
+
+	access, infoUser, err := b.broker.IsAuthorized(context.Background(), sessionID, authenticationData)
+	if err != nil {
+		return "", "", dbus.MakeFailedError(err)
+	}
+	return access, infoUser, nil
+}
+
+// EndSession calls the broker mock corresponding method and returns the result or an error if requested.
+func (b *BrokerBusMock) EndSession(sessionID string) (dbusErr *dbus.Error) {
+	if sessionID == "ES_bus_error" {
+		return dbus.MakeFailedError(errors.New("erroring out as requested"))
+	}
+
+	err := b.broker.EndSession(context.Background(), sessionID)
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+	return nil
+}
+
+// CancelIsAuthorized calls the broker mock corresponding method.
+func (b *BrokerBusMock) CancelIsAuthorized(sessionID string) (dbusErr *dbus.Error) {
+	b.broker.CancelIsAuthorized(context.Background(), sessionID)
+	return nil
+}
+
+/* BrokerMock implementation */
+
+// NewSession returns default values to be used in tests or an error if requested.
+func (b *BrokerMock) NewSession(ctx context.Context, username, lang string) (sessionID, encryptionKey string, err error) {
+	if username == "NS_error" {
```

Comment:
and that way, we remove also all those special cases, only one "error returned from the bus blackbox"